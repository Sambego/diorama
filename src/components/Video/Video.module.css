.container {
  width: 80vw;
  height: calc((80vw / 16) * 9);
}

.full {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 0;

  width: 100vw;
  height: 100vh;
  object-fit: cover;

  video {
    width: 100vw;
    height: 100vh;
    object-fit: cover;
  }
}

.overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;

  opacity: 0.6;
}
