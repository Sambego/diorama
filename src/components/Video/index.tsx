import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./Video.module.css";

interface VideoProps {
	src: string;
	className?: string;
	style?: React.CSSProperties;
	color?: React.CSSProperties["backgroundColor"];
	loop?: boolean;
	autoplay?: boolean;
	full?: boolean;
}
function Video({
	src,
	color,
	full,
	autoplay,
	loop,
	style,
	className,
}: VideoProps) {
	const classes = classnames(
		styles.video,
		{
			[styles.full]: full,
		},
		className,
		"diorama-video-container"
	);

	return (
		<div className={classes} style={style}>
			<video
				src={src}
				autoPlay={autoplay}
				loop={loop}
				muted
				className="diorama-video-player"
			/>
			{color ? (
				<div
					className={`${styles.overlay} diorama-video-overlay`}
					style={{
						backgroundColor: color,
					}}
				/>
			) : null}
		</div>
	);
}

Video.propTypes = {
	autoplay: PropTypes.bool,
	loop: PropTypes.bool,
	full: PropTypes.bool,
	color: PropTypes.string,
	style: PropTypes.shape({}),
	src: PropTypes.string.isRequired,
	className: PropTypes.string,
};

Video.defaultProps = {
	autoplay: false,
	loop: false,
	full: false,
	color: undefined,
	style: {},
	className: "",
};

export default Video;
