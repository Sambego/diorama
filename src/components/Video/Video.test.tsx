import React from 'react';
import renderer from 'react-test-renderer';
import Video from ".";

it('renders correctly', () => {
  const tree = renderer.create(<Video src="#" />).toJSON();
  expect(tree).toMatchSnapshot();
});
