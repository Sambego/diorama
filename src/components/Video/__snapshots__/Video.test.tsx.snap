// Vitest Snapshot v1

exports[`renders correctly 1`] = `
<div
  className="diorama-video-container"
  style={{}}
>
  <video
    autoPlay={false}
    className="diorama-video-player"
    loop={false}
    muted={true}
    src="#"
  />
</div>
`;
